### ISSUE TITLE
Singularity crashed on durationMillis being negative

### ISSUE THREAD

**Issue by grepsr** [HubSpot/Singularity#1251]
My singularity crashed today with an UNRECOVERABLE_ERROR.

Not sure what caused it. After it restarted, it was all good. Trace below:

```
java.lang.IllegalArgumentException: durationMillis must not be negative

ERROR [2016-08-26 06:27:04,236] com.hubspot.singularity.mesos.SingularityMesosScheduler: Received fatal error while accepting offers - will decline all available offers
! java.lang.IllegalArgumentException: durationMillis must not be negative
! at org.apache.commons.lang3.Validate.inclusiveBetween(Validate.java:969) ~[SingularityService.jar:0.8.0]
! at org.apache.commons.lang3.time.DurationFormatUtils.formatDuration(DurationFormatUtils.java:131) ~[SingularityService.jar:0.8.0]
! at org.apache.commons.lang3.time.DurationFormatUtils.formatDuration(DurationFormatUtils.java:114) ~[SingularityService.jar:0.8.0]
! at com.hubspot.mesos.JavaUtils.duration(JavaUtils.java:115) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorManager.log(CuratorManager.java:67) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorAsyncManager.queryAndReturnResultsThrows(CuratorAsyncManager.java:308) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorAsyncManager.getAsyncThrows(CuratorAsyncManager.java:100) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorAsyncManager.getAsyncChildrenThrows(CuratorAsyncManager.java:52) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorAsyncManager.getAsyncChildren(CuratorAsyncManager.java:236) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.TaskManager.getPendingTasks(TaskManager.java:605) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.scheduler.SingularityScheduler.getDueTasks(SingularityScheduler.java:301) ~[SingularityService.jar:0.8.0]
! at com.palominolabs.metrics.guice.TimedInterceptor.invoke(TimedInterceptor.java:47) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.mesos.SingularityMesosScheduler.resourceOffers(SingularityMesosScheduler.java:151) ~[SingularityService.jar:0.8.0]
! at com.palominolabs.metrics.guice.TimedInterceptor.invoke(TimedInterceptor.java:47) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.mesos.SingularityMesosSchedulerDelegator.resourceOffers(SingularityMesosSchedulerDelegator.java:175) [SingularityService.jar:0.8.0]
ERROR [2016-08-26 06:27:04,237] com.hubspot.singularity.mesos.SingularityMesosSchedulerDelegator: Scheduler threw an uncaught exception - exiting
! java.lang.IllegalArgumentException: durationMillis must not be negative
! at org.apache.commons.lang3.Validate.inclusiveBetween(Validate.java:969) ~[SingularityService.jar:0.8.0]
! at org.apache.commons.lang3.time.DurationFormatUtils.formatDuration(DurationFormatUtils.java:131) ~[SingularityService.jar:0.8.0]
! at org.apache.commons.lang3.time.DurationFormatUtils.formatDuration(DurationFormatUtils.java:114) ~[SingularityService.jar:0.8.0]
! at com.hubspot.mesos.JavaUtils.duration(JavaUtils.java:115) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorManager.log(CuratorManager.java:67) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorAsyncManager.queryAndReturnResultsThrows(CuratorAsyncManager.java:308) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorAsyncManager.getAsyncThrows(CuratorAsyncManager.java:100) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorAsyncManager.getAsyncChildrenThrows(CuratorAsyncManager.java:52) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.CuratorAsyncManager.getAsyncChildren(CuratorAsyncManager.java:236) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.data.TaskManager.getPendingTasks(TaskManager.java:605) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.scheduler.SingularityScheduler.getDueTasks(SingularityScheduler.java:301) ~[SingularityService.jar:0.8.0]
! at com.palominolabs.metrics.guice.TimedInterceptor.invoke(TimedInterceptor.java:47) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.mesos.SingularityMesosScheduler.resourceOffers(SingularityMesosScheduler.java:151) ~[SingularityService.jar:0.8.0]
! at com.palominolabs.metrics.guice.TimedInterceptor.invoke(TimedInterceptor.java:47) ~[SingularityService.jar:0.8.0]
! at com.hubspot.singularity.mesos.SingularityMesosSchedulerDelegator.resourceOffers(SingularityMesosSchedulerDelegator.java:175) ~[SingularityService.jar:0.8.0]
ERROR [2016-08-26 06:27:04,238] com.hubspot.singularity.SingularityAbort: Singularity on abc.com is aborting due to UNRECOVERABLE_ERROR
INFO  [2016-08-26 06:27:04,238] com.hubspot.singularity.SingularityAbort: Attempting to flush logs and wait 00:00.100 ...
I0826 06:27:04.381853    36 sched.cpp:1937] Asked to abort the driver
```

**Comment by ssalinas** [comment]
Thought this was the same issue as #1174 at first, but looks to be in a different spot. Not sure how the value came up negative here, but I'll take a closer look.

**Comment by ssalinas** [comment]
Fixed by #1252
